File: kool_image.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <fcntl.h>
#include <unistd.h>

int main(){
  int file = open("kool_image.ppm", O_CREAT | O_WRONLY, 0777);
  int xres = 500;
  int yres = 500;
  int max_color = 255;

  char* buffer = malloc(256*sizeof(char));
  sprintf(buffer,"P3 %i %i %i", xres, yres, max_color);
  write( file, buffer, sizeof(buffer));

  int x,y, red, green, blue;
  green = 0;

  for(x = 0; x < xres; x++){
    for(y= 0; y < yres; y++){
      red = x % max_color;
      blue = y % max_color;
      
      sprintf (buffer, " %d %d %d ", red, green, blue);
      printf("%s\n", buffer);
      write (file, buffer, sizeof(buffer));
    }
  }
  return 0;
}

File: hw0/kool_image.py
def kool_image():
    f = open("kool_image.ppm", "w")
    f.write("P3 250 250 255 ")
    red = 0
    green = 0
    blue = 0
    while (blue < 250):
        green = 0
        while (green < 250):
            f.write( "%d %d %d " % ( red, green, blue ) )
            green += 1
        blue += 1
    f.close()

if __name__=='__main__':
    kool_image()
